test/prototype/test_bitpacking.py

```diff
@@ -0,0 +1,52 @@
+import torch
+from torchao.prototype.common.bitpacking import pack, unpack
+import pytest
+
+
+if not TORCH_VERSION_AFTER_2_4:
+    pytest.skip("Unsupported PyTorch version")
+
+def test_uint4_to_uint8():
+    test_tensor = torch.randint(0, 15, (4, 4), dtype=torch.uint8)
+    packed = pack(test_tensor, 8, 4)
+    unpacked = unpack(packed, 4)
+    unpadded = unpacked[:test_tensor.shape[0], ...]
+    assert(unpadded.allclose(test_tensor))
+    
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
+@pytest.mark.skipif(not has_triton(), reason="unsupported without triton")
+def test_uint4_to_uint8_compile():
+    torch._dynamo.config.specialize_int = True
+    pack_compiled = torch.compile(pack)
+    unpack_compiled = torch.compile(unpack)
```

Comment:
can you test `torch.compile(unpack, fullgraph=True)` that'll ensure only 1 kernel is ever generated